File: app/src/main/java/vn/loitp/app/activity/customviews/viewpager/detectviewpagerswipeout/DetectViewPagerSwipeOutActivity.java
package vn.loitp.app.activity.customviews.viewpager.detectviewpagerswipeout;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.core.base.BaseFontActivity;
import com.core.base.BaseFragment;
import com.core.utilities.LLog;
import com.core.utilities.LStoreUtil;
import com.core.utilities.LUIUtil;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import vn.loitp.app.R;

public class DetectViewPagerSwipeOutActivity extends BaseFontActivity {
    private List<VPPhoto> vpPhotoList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ViewPager viewPager = findViewById(R.id.viewpager);

        int max = 3;
        for (int i = 0; i < max; i++) {
            VPPhoto vpPhoto = new VPPhoto();
            vpPhoto.setColor(LStoreUtil.INSTANCE.getRandomColor());
            vpPhoto.setString("Page " + i + "/" + (max - 1));
            vpPhotoList.add(vpPhoto);
        }

        LUIUtil.INSTANCE.setPullLikeIOSHorizontal(viewPager, new LUIUtil.Callback() {
            @Override
            public void onUpOrLeft(float offset) {
                LLog.d(getTAG(), "onUpOrLeft " + offset);
                showShort("Detect Left");
            }

            @Override
            public void onUpOrLeftRefresh(float offset) {
                LLog.d(getTAG(), "onUpOrLeftRefresh " + offset);
            }

            @Override
            public void onDownOrRight(float offset) {
                LLog.d(getTAG(), "onDownOrRight " + offset);
                showShort("Detect Right");
            }

            @Override
            public void onDownOrRightRefresh(float offset) {
                LLog.d(getTAG(), "onDownOrRightRefresh " + offset);
            }
        });

        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(adapter);

    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_viewpager_detect_swipeout;
    }

    private class ViewPagerAdapter extends FragmentStatePagerAdapter {

        ViewPagerAdapter(FragmentManager fm) {
            super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        }

        @NotNull
        @Override
        public Fragment getItem(int position) {
            Bundle bundle = new Bundle();
            bundle.putSerializable("vpphoto", vpPhotoList.get(position));
            BaseFragment fragment = null;
            fragment = new FrmPhoto();
            fragment.setArguments(bundle);
            return fragment;
        }

        @Override
        public int getCount() {
            return 3;
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/viewpager/detectviewpagerswipeout/DetectViewPagerSwipeOutActivity.kt
package vn.loitp.app.activity.customviews.viewpager.detectviewpagerswipeout

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LStoreUtil
import com.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_view_pager_detect_swipe_out.*
import vn.loitp.app.R
import java.util.*

@LogTag("DetectViewPagerSwipeOutActivity")
@IsFullScreen(false)
class DetectViewPagerSwipeOutActivity : BaseFontActivity() {
    private val vpPhotoList: MutableList<VPPhoto> = ArrayList()

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_view_pager_detect_swipe_out
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        val max = 3
        for (i in 0 until max) {
            val vpPhoto = VPPhoto()
            vpPhoto.color = LStoreUtil.randomColor
            vpPhoto.string = "Page " + i + "/" + (max - 1)
            vpPhotoList.add(vpPhoto)
        }
        LUIUtil.setPullLikeIOSHorizontal(
            viewPager = viewPager,
            onUpOrLeft = { offset ->
                logD("onUpOrLeft $offset")
                showShortInformation("Detect Left")
            },
            onUpOrLeftRefresh = { offset ->
                logD("onUpOrLeftRefresh $offset")
            },
            onDownOrRight = { offset ->
                logD("onDownOrRight $offset")
                showShortInformation("Detect Right")
            },
            onDownOrRightRefresh = { offset ->
                logD("onDownOrRightRefresh $offset")
            }
        )
        val adapter = ViewPagerAdapter(supportFragmentManager)
        viewPager.adapter = adapter
    }

    private inner class ViewPagerAdapter(fm: FragmentManager) :
        FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

        override fun getItem(position: Int): Fragment {
            val bundle = Bundle()
            bundle.putSerializable("vpphoto", vpPhotoList[position])
            val fragment = FrmPhoto()
            fragment.arguments = bundle
            return fragment
        }

        override fun getCount(): Int {
            return 3
        }
    }
}
